pkg/component/controller/cplb/cplb_reconciler.go

```diff
@@ -109,22 +123,154 @@ func (r *CPLBReconciler) maybeUpdateIPs(endpoint *corev1.Endpoints) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	// endpoints are not guaranteed to be sorted by IP address
-	slices.Sort(newAddresses)
+	for _, addr := range newAddresses {
+		if _, ok := r.healthCheckers[addr]; !ok {
+			r.healthCheckers[addr] = r.newHealthChecker(ctx, restConfig, addr, nil)
+		}
+	}
 
-	if !slices.Equal(r.addresses, newAddresses) {
-		r.addresses = newAddresses
-		r.log.Infof("Updated the list of IPs: %v", r.addresses)
-		select {
-		case r.updateCh <- struct{}{}:
-		default:
+	shouldNotify := false
+	for addr := range r.healthCheckers {
+		if !slices.Contains(newAddresses, addr) {
+			r.healthCheckers[addr].Stop()
+			delete(r.healthCheckers, addr)
+			shouldNotify = true
 		}
 	}
+	if shouldNotify {
+		r.updateCh <- struct{}{}
+	}
 }
 
 // GetIPs gets a thread-safe copy of the current list of IP addresses
 func (r *CPLBReconciler) GetIPs() []string {
 	r.mu.Lock()
 	defer r.mu.Unlock()
-	return slices.Clone(r.addresses)
+
+	var healthyAddrs []string
+	for addr := range r.healthCheckers {
+		if r.healthCheckers[addr].healthy {
+			healthyAddrs = append(healthyAddrs, addr)
+		}
+	}
+
+	if len(healthyAddrs) == 0 && len(r.healthCheckers) > 0 {
+		r.log.Warn("No healthy API servers detected by CPLBReconciler, returning every known address")
+		return slices.Collect(maps.Keys(r.healthCheckers))
+	}
+	return healthyAddrs
+}
+
+// healthChecker provides health checking functionality for Kubernetes API servers
+type healthChecker struct {
+	cancelFunc    context.CancelFunc
+	rc            rest.Interface
+	log           *logrus.Entry
+	healthy       bool
+	updateCh      chan<- struct{}
```

Comment:
This somehow looks like it could be replaced by

```suggestion
	healthy       value.Latest[bool]
```